### PR TITLE
Company register

### PR DESCRIPTION
**Pull request by meriem26**

### CODE REVIEW

File: client/src/components/CompanyRegister.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+              name: "",
+              description: "",
+              industry: [],
+              email: "",
+              password: ""
+            },
+            success: true
+          });
+        } else {
+          this.setState({});
```
Comment (kabaros) [inline client/src/components/CompanyRegister.js]: this means that if there `data.success` is `false`, then reset the form? Is that the behaviour you're after?

File: client/src/components/CompanyRegister.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    const property = event.target.name;
+    const value = event.target.value;
+    this.setState(function(prevState) {
+      const newEntries = prevState.companyProfile;
+      newEntries[property] = value;
+      return { companyProfile: newEntries };
+    });
+  };
+  render() {
+    console.log(this.state.companyProfile);
```
Comment (kabaros) [inline client/src/components/CompanyRegister.js]: remove console.log

File: client/src/components/CompanyRegister.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        }
+        this.resetForm(data.success);
+      })
+      .catch(err => {
+        this.setState({ success: false, error: err });
+      });
+  };
+
+  handleIndustry = (event, data) => {
+    const selectedIndustries = data.value;
```
Comment (kabaros) [inline client/src/components/CompanyRegister.js]: just for you to know, you could use destructuring to write this line as such `const { value: selectedIndustries} = data` ... this tells to `destructure` _value_ from _data_ object then rename it to _selectedIndustries_  ... you can read more about this here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment


you don't need to implement it like this .. but it's something in ES6 and it's good to learn

File: client/src/components/CompanyRegister.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        description: "",
+        industry: [],
+        email: "",
+        password: ""
+      },
+      success: success
+    });
+  };
+  handleChange = event => {
+    const property = event.target.name;
```
Comment (kabaros) [inline client/src/components/CompanyRegister.js]: same as previous comment .. you can write with destructuring syntax like 
```
const { target } = event; // destructuring target from event
const { name: property } = target // destrucuring name and renaming it to property
```

More info about destructuring - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

File: client/src/components/CompanyRegister.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+                    required
+                    placeholder="Select Industry"
+                    onChange={this.handleIndustry}
+                  />
+                </Form.Field>
+                <Button
+                  type="submit"
+                  primary
+                  fluid
+                  // onClick={this.state.success === true ? this.resetForm : null}
```
Comment (kabaros) [inline client/src/components/CompanyRegister.js]: remove commented out code

File: client/src/components/CompanyRegister.js [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    error: null,
+    success: null
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+
+    companyRegister(this.state.companyProfile)
+      .then(data => {
+        console.log(data);
```
Comment (kabaros) [inline client/src/components/CompanyRegister.js]: remove console.log

File: client/src/components/CompanyRegister.js [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+            },
+            success: true
+          });
+        } else {
+          this.setState({});
+        }
+        this.resetForm(data.success);
+      })
+      .catch(err => {
+        this.setState({ success: false, error: err });
```
Comment (khaledkz) [inline client/src/components/CompanyRegister.js]: it's better to return a string (currently you returning an object), for example, 
```this.setState({ err.message ||  'error' });```

File: client/src/components/CompanyRegister.js [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+            </Grid.Row>
+            <Grid.Row width={15}>
+              {this.state.success === true && (
+                <Message positive>
+                  <Message.Header>
+                    Company Profile Created successfully
+                  </Message.Header>
+                </Message>
+              )}
+              {this.state.success === false && (
```
Comment (khaledkz) [inline client/src/components/CompanyRegister.js]: Do you need to check if there is an error as well?
so you should check if success is false or if the error is true

File: client/src/components/CompanyRegister.js (hunk 1 above)
Comment (meriem26) [reply client/src/components/CompanyRegister.js]: data.success should be true to reset the form, if it's false then it will return am error message .

**Pull request merged**